crates/ruff/src/rules/pyupgrade/rules/format_literals.rs

```diff
@@ -16,14 +16,16 @@ use crate::rules::pyflakes::format::FormatSummary;
 #[violation]
 pub struct FormatLiterals;
 
-impl AlwaysAutofixableViolation for FormatLiterals {
+impl Violation for FormatLiterals {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
```

Comment:
I only changed the rules to Violation, because they don't always provide a fix (I found out about this by the added assertions in the tests). That's why Sometimes is correct(or we have to change the implementation to always emit a fix)

Reply:
Ahh right right. I didn't realize that.